app/models/evss_claim_document.rb

```diff
@@ -1,17 +1,24 @@
 # frozen_string_literal: true
 require 'common/models/base'
+require 'iconv'
+require 'pdf/reader'
+require 'rchardet'
 
 class EVSSClaimDocument < Common::Base
   include ActiveModel::Validations
+  include ActiveModel::Validations::Callbacks
 
   attribute :evss_claim_id, Integer
   attribute :tracked_item_id, Integer
   attribute :document_type, String
   attribute :file_name, String
+  attribute :file_obj
 
-  validates(:tracked_item_id, presence: true)
   validates(:file_name, presence: true)
   validate :known_document_type?
+  validate :unencrypted_pdf?
+  before_validation :normalize_text
+  before_validation :normalize_tracked_item_id
```

Comment:
I don't think we can use a setter for normalize_text because it's doing double duty: It returns false if the text can't be converted, which makes the validation fail. We'd need to add the validation test back in somehow if we got rid of that.